### DISCUSSION TITLE
Overwriting public property model from browser

### DISCUSSION THREAD

**Discussion by torian257x** [livewire/livewire#3956] [1. Help]
I noticed you can overwrite

```
class MyComp extends Component{
public $product_id = 1;

}
```

in the browser you can do

`Livewire.first().product_id = 2`

to change the product_id to 2

What if I had an entire model:

```
class MyComp extends Component{
public ?Product $product = null;

 mount(){
$this->product = Product::find(1);
}
```

What can I do to overwrite `$this->product`  with id 1

to

`$this->product` with id 2?

the reason I'm asking is not "for fun". I am actually worried that clients can somehow overwrite entire model properties.

So I really look forward to an answer for this one.

**Comment by joshhanley** [comment]
@torian257x yeah that's certainly an understandable concern. Livewire already has protection built in for that, where if your property is an eloquent model, or an eloquent collection, those properties can't be modified from the front end unless there is a rule in your rules array for each specific attribute on a model.

Have a look at this section of the docs https://laravel-livewire.com/docs/2.x/properties#binding-models

> Note: For this to work, you have a validation entry in the `$rules` property for any model attributes you want to bind to. Otherwise, an error will be thrown.

For example, if you tried to edit `Livewire.first().product.id = 2` you will get an error, unless you have this in your rules array:

```php
protected $rules = [
    'product.id' => 'nullable',
];
```

Once a rule exists for `product.id` then only the `id` attribute of the `product` property can be modified but none of the rest of the attributes.

I'd also recommend running validation and adding a check to your validation to ensure that the ID that they change to is both valid and a model ID that they are allowed to access.

Hope this helps!

**Reply by torian257x** [reply]
thank you, I saw that passage... what I wonder is whether you can update the entire product object to be something else? I see they limited the attributes of the model. But can't you just _somehow_ overwrite the entire model itself?

e.g. `Livewire.first().product = {id: 1, name: 'car tire', price: 15}`

**Reply by joshhanley** [reply]
@torian257x yeah same thing, it should be blocked, unless you have a rule for the product as a whole like `'product' => 'nullable'`, but only due to it being an eloquent model. Any other properties (like int, string, etc) can be modified without a rule.

**Reply by torian257x** [reply]
ok. Thus array, stdClass can be completely overwritten as well? would be good to have that a bit more exposed in the docs.

**Reply by joshhanley** [reply]
Array - yep, but stdClass - no, as it isn't supported by default as a Livewire property type. Yeah agreed!
